packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -2352,6 +2359,9 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
         }
       });
       hideToolbar();
+      if (shouldSelectWordEdgeAfterReplacement) {
+        renderEditable.selectWordEdge(cause: cause);
+      }
```

Comment:
Would it work out if it were the caller's responsibility to call selectWordEdge after calling this if they want that behavior?

Reply:
I mean as long as they call it, it should be fine. Otherwise, the word would stay highlighted after the replacement was made until another tap was made.